### PR TITLE
Implement the 'add event' functionality in batch edit + test

### PR DESCRIPTION
**Pull request by bp323**

### CODE REVIEW

**Comment by bp323** [comment]
Reassigning to @Coenego for review.

File: apps/timetable/admin/index.html [hunk 1]
```diff
@@ -135,9 +135,16 @@
...
             <%= _.partial('admin-batch-edit', {
-                    'gh': gh,
-                    'data': data
-                }) %>
+                'gh': gh,
+                'data': data
+            }) %>
+        </script>
+
+        <script id="gh-batch-edit-event-row-template" type="text/template">
```
Comment (Coenego) [inline apps/timetable/admin/index.html]: Can you just add a comment above the script tag to be consistent with the other templates?

File: tests/qunit/tests/js/api.event.js [hunk 2]
```diff
@@ -114,6 +114,114 @@ require(['gh.core', 'gh.api.tests'], function(gh, testAPI) {
...
+    QUnit.asyncTest('createEvent', function(assert) {
+        expect(20);
+
+        // Fetch a random test app
+        var app = testAPI.getTestApp();
+
+        // Verify that an error is thrown when no callback was provided
+        assert.throws(function() {
+            gh.api.eventAPI.createEvent('displayName', '2014-12-31', '2015-01-01', 'description', null, 'location', 'notes', null, null, null);
+        }, 'Verify that an error is thrown when an invalid callback was provided');
```
Comment (Coenego) [inline tests/qunit/tests/js/api.event.js]: when no callback was provided

File: shared/gh/js/controller/gh.admin-batch-edit.js [hunk 3]
```diff
@@ -30,6 +30,28 @@ define(['gh.api.series', 'gh.api.util', 'gh.api.event', 'gh.admin-constants'], f
     };
 
     /**
+     * Add a new event row to the table and initialise the editable fields in it
+     */
```
Comment (Coenego) [inline shared/gh/js/controller/gh.admin-batch-edit.js]: `@private` missing

**Comment by Coenego** [comment]
General question: When creating new events, should we limit the entry for minutes to intervals of 15 minutes? (cc @oszkarnagy)

**Comment by Coenego** [comment]
Left some minor comments and a question. Reassigning to @bp323

**Comment by Coenego** [comment]
Looks good. I'll go ahead and merge this.

**Pull request merged**